### ISSUE TITLE
Error while creating a page under a folder.

### ISSUE THREAD

**Issue by zouloux** [getgrav/grav-plugin-admin#1852]
Hi all,
I currently have a problem when creating a new page directly under a folder.

How to reproduce : 
- Connect to admin
- Select "pages"
- Create a new folder
- Click on "Add", select previously created folder as "parent page"
- Continue
- Save page, then this error shows up : 

```
TypeError
Argument 1 passed to Grav\Common\Page\Page::move() must implement interface Grav\Common\Page\Interfaces\PageInterface, null given, called in /Users/zouloux/Documents/local/sennesro/trunk/dist/user/plugins/admin/classes/plugin/AdminController.php on line 1325
```

**Stack** :
Grav 1.7.0-rc.7
Grav Admin v1.10.0-rc.3

Thanks a lot !

**Comment by mahagr** [comment]
Reproduced the issue. It can also be reproduced using Grav 1.6 and Admin 1.9.

**Comment by mahagr** [comment]
Also, this bug happens when using the old pages admin, not with Flex pages.

**Comment by zouloux** [comment]
Yep, forgot to mention, thanks ! All my themes are not "flex-page ready" yet.
Do you have any idea where does it comes from ?

**Comment by mahagr** [comment]
Right now we're not recommending using Flex in the frontend yet -- the feature is still experimental and missing some minor details.

That said, admin uses flex by default and you need to manually turn it off. The implementation is considered to be complete, but more testing is welcome. It also should work with all the themes and most plugins.

**Comment by zouloux** [comment]
Yes but here the issue is with Flex disabled (regular pages system) and is reproducible on RC but also in stable.

**Comment by mahagr** [comment]
Yes, I had the same conclusion. It looks like the folder is not considered to be an existing page in the old logic, which causes this issue.

**Comment by zouloux** [comment]
Hi, do you have any clue to give me on this ? I can try to make a PR when I got some time.

**Comment by zouloux** [comment]
Ok so I think I got it, it's what I thought. The new UI for page selection give a link from `root` but `page->dispatch` need to be from `/user/pages`.
Just added a failsafe, I think it has no side effects other than if you name your folders user and pages ;)

Added this line 1298 in `AdminController.php`, before `$route = '/' ...`
```
        // Remove /user/pages given by page finder
        $route = ltrim($route, '/user/pages');
```

Making a PR :)

**Comment by zouloux** [comment]
I can't propose a PR.

**Comment by zouloux** [comment]
Ok so moving an existing page with the "move" button crashes too.
Only when the destination is a folder, and not a page.
The folder path is rendered like so : 
`/user/pages/05.test`
and with a page it works because it's like :
`user/pages/test`.
If anybody can help me on this, we just need to detect folder vs pages, and convert to page route. Thanks !

**Comment by rhukster** [comment]
I'm sure @mahagr will jump back on when he's feeling better. he's unavailable with the flu currently.

**Comment by zouloux** [comment]
Sure, nothing important, just trying to get it done myself. Take care @mahagr, sorry for the useless flood !

**Comment by mahagr** [comment]
@zouloux Don't use trim like that, it just removes all the individual characters you list in there, so the route will lose a few letters from the beginning.

**Comment by zouloux** [comment]
@mahagr Ooooh I see 😅 I will replace by, `strpos` + `substr` but you get the idea. Thanks for your feedback !

**Comment by zouloux** [comment]
This do the trick, add this in AdminController.php line 1298 :
```
        // Remove /user/pages given by page finder
        $pathToRemove = '/user/pages';
        if (strpos($route, $pathToRemove) === 0)
            $route = substr($route, strlen($pathToRemove), strlen($pathToRemove));
```
The create new page in specific folder works BUT, like I said [here](https://github.com/getgrav/grav-plugin-admin/issues/1852#issuecomment-607341412), the move my page to another folder from admin does not works yet. This time, the `$route` given is not from `/user/pages` but folders are named like file system, not route system ( ex : `/sub-folder/` is `/01.sub-folder` ), so, same error on `Page->move(null)`. Any idea on this ? It looks like the finder-like module gives wrong routes. Thanks a lot.

**Comment by zouloux** [comment]
In fact even when creating a new page in a sub-folder, i missed it but the URL is not valid (has user/pages in it). Replaced `type: parents` by good old `type: pages` in `new.yaml` and my pages blueprints. Now I got the old pages selector, I'm pretty sure this has been replaced by the new one for performances purpose (to allow virtually infinite number of pages). But this new page selector seems not ready yet for old pages system (not the flex one). I hope I gave enough informations on this, I got a working patch on my side but it may be weird for new comers.

**Comment by zouloux** [comment]
You know what ? It seems to work fine !
My Grav Install crashed hard today. I needed to re-install everything, so I'm on testing channel, latest Grav RC and Admin panel 1.10.0 RC 10. Old pages system but flex object installed and working. I can now create new pages under a folder (with the new page selector component). Moving the page works well too (component is the old page selector). Closing it, thanks !
